may-2018-reenabled-opcodes.md

```diff
@@ -0,0 +1,415 @@
+re-designed or new opcodes have been added to address specific issues.
+
+This document contains the specifications for the opcodes that are to be added in the May 2018 protocol upgrade. We 
+anticipate that additional opcodes will be proposed for the November 2018, or later, protocol upgrades.
+
+The opcodes that are to be added are:
+
+|Word       |OpCode |Hex |Input         |Output  | Description                                                      |
+|-----------|-------|----|--------------|--------|------------------------------------------------------------------|
+|OP_CAT     |126    |0x7e|x1 x2         |out     |Concatenates two byte arrays                                      |
+|OP_SPLIT   |127    |0x7f|x n           |x1 x2   |Split byte array *x* at position *n*                              |
+|OP_AND     |132    |0x84|x1 x2         |out     |Boolean *AND* between each bit of the inputs                      |
+|OP_OR      |133    |0x85|x1 x2         |out     |Boolean *OR* between each bit of the inputs                       |
+|OP_XOR     |134    |0x86|x1 x2         |out     |Boolean *EXCLUSIVE OR* between each bit of the inputs             |
+|OP_DIV     |150    |0x96|a b           |out     |*a* is divided by *b*                                             |
+|OP_MOD     |151    |0x97|a b           |out     |return the remainder after *a* is divided by *b*                  |
+|OP_NUM2BIN |128    |0x80|a b           |out     |convert numeric *a* into byte array of length *b*                 |
+|OP_BIN2NUM |129    |0x81|x             |out     |convert byte array *x* into numeric                               |
+
+Splice operations: `OP_CAT`, `OP_SPLIT`**
+
+Bitwise logic: `OP_AND`, `OP_OR`, `OP_XOR`
+
+Arithmetic: `OP_DIV`, `OP_MOD`
+
+New operations: 
+* `x OP_BIN2NUM -> n`, convert a binary array `x` into a valid (canonical) numeric element
+* `n m OP_NUM2BIN -> out`, convert a numeric value `n` into a byte array of length `m`
+
+Further discussion of the purpose of these new operations can be found below under *bitwise operations*.
+
+** A new operation, `OP_SPLIT`, has been designed as a replacement for `OP_SUBSTR`, `OP_LEFT`and `OP_RIGHT`. 
+The original operations can be implemented with varying combinations of `OP_SPLIT`, `OP_SWAP` and `OP_DROP`.
+
+
+## <a name="data-types"></a>Script data types
+
+It should be noted that in script operation data values on the stack are interpreted as either binary strings (i.e. an array of bytes) 
+or numbers.  **All data on the stack is interpreted as an array of bytes unless specifically stated as being interpreted as numeric.**
+
+The numeric type has specific limitations:
+1. The used encoding is little endian with an explicit sign bit (the highest bit of the last byte).
+2. They cannot exceed 4 bytes in length.
+3. They must be encoded using the shortest possible byte length (no zero padding)
```

Comment:
In the OP_MOD, OP_DIV code this still seems to be dependent on SCRIPT_VERIFY_MINIMALDATA, which is determines standardness? Or did I miss it?